### ISSUE TITLE
case sensitive error when load ConfigParser or else

### ISSUE THREAD

**Issue by maiconfaria** [TRON-Bioinformatics/EasyFuse#17]
https://github.com/TRON-Bioinformatics/EasyFuse/blob/710ae510cc09484cf9ae5fa5bf171896bcaf5692/processing.py#L15

Simple test with singularity run easyfuse.simg python /code/easyfuse/processing.py
Traceback (most recent call last):
File "/code/easyfuse/processing.py", line 15, in <module>
from configparser import ConfigParser
ImportError: No module named configparser

**Comment by ibn-salem** [comment]
Dear @maiconfaria ,

this issue should be fixed with the latest version. Did you tried it with the docker image version 1.3.6?

Thanks, 
Jonas

**Comment by maiconfaria** [comment]
I was using the latest this Friday.

**Comment by patricksorn** [comment]
Hi @maiconfaria ,

I think your `latest` tag might be outdated. 
Can you try printing out the EasyFuse version by using `/code/easyfuse/processing.py --version` within your container? 
Could you also try loading the specific version tag (1.3.6) instead of using `latest` using the following command:
singularity exec --bind <your_reference_folder>:/ref --bind <your_data_folder>:/data docker://docker.io/tronbioinformatics/easyfuse:1.3.6 python /code/easyfuse/processing.py -i /code/easyfuse/test_case/ -o /data/results/

Best regards,
Patrick

**Comment by maiconfaria** [comment]
Thanks for the answer. Here is what I get in terms of version:

```
maiconfaria@maiconHPCNow:~$ singularity pull docker://tronbioinformatics/easyfuse:1.3.6
INFO:    Using cached SIF image
maiconfaria@maiconHPCNow:~$ singularity shell easyfuse_
easyfuse_1.3.6.sif        easyfuse_latest_orig.sif  easyfuse_latest.sif       
maiconfaria@maiconHPCNow:~$ singularity shell easyfuse_1.3.6.sif 
Singularity> /code/easyfuse/processing.py --version
usage: EasyFuse [-h] -i INPUT_PATHS [INPUT_PATHS ...] -o OUTPUT_FOLDER
                [-c CONFIG_FILE] [--tool_support TOOL_SUPPORT]
                [-p JOBNAME_SUFFIX] [--version VERSION]
EasyFuse: error: argument --version: expected one argument
```

**Comment by patricksorn** [comment]
Sorry for the inconvenience! There must be some bug in the version info. Will include a fix to this in the next revision.
Can you in the meantime do `cat /code/easyfuse/config.ini` ?
Or just try to run it with the test_case:
`python /code/easyfuse/processing.py -i /code/easyfuse/test_case/ -o /data/results/`

**Comment by maiconfaria** [comment]
The version seems to be the 1.3.6

```
Singularity> cat /code/easyfuse/config.ini
# Required user input:
# 1) Which fusion prediction tools should be executed (tools)
# 2) Which post-processing steps should be executed (fd_tools)
# 3) Which reference data shall be used (ref_trans_version & ref_genome_build)
# 4) To whom shall slurm mails be sent to (receiver)

[general]
version = 1.3.6
pipeline_name = EasyFuse
tools = QC,Readfilter,Fusioncatcher,Star,Starfusion,Infusion,Mapsplice,Soapfuse,Fetchdata,Summary
fusiontools = Fusioncatcher,Starfusion,Infusion,Mapsplice,Soapfuse
fd_tools = Fusiongrep,Contextseq,Starindex,ReadFilter2,ReadFilter2b,StaralignBest,BamindexBest,RequantifyBest
sender = sender@mail.com
receiver = receiver@mail.com
min_read_len_perc = 0.75
max_dist_proper_pair = 200000
cis_near_distance = 1000000
model_pred_threshold = 0.5
tsl_filter = 4,5,NA
requant_mode = best
context_seq_len = 400
ref_genome_build = hg38
ref_trans_version = ensembl
queueing_system = none
time_limit = 30-00:00:0
partition = Compute
user = sorn
module_dir = /code/easyfuse/

# Define ressource usage (cpu (number of threads), mem (ram in Gb)):
[resources]
qc = 6,10
readfilter = 6,50
star = 6,40
mapsplice = 6,30
fusioncatcher = 6,30
starchip = 12,30
starfusion = 6,30
infusion = 6,30
soapfuse = 6,20
classification = 1,16
fetchdata = 12,50
summary = 1,16


# execution command for individual programs (what you write here should be identical to what is typed in the console)
[commands]
# for qc
fastqc = fastqc
skewer = /code/skewer-0.2.2/skewer
# for processing
mapsplice = mapsplice.py
fusioncatcher = /code/fusioncatcher/1.0/bin/fusioncatcher
starfusion = STAR-Fusion
infusion = /code/InFusion-0.8/infusion
soapfuse = /code/SOAPfuse/1.27/SOAPfuse-RUN.pl
# for processing and fetch data
star = STAR
samtools = samtools
# for liftover
crossmap = test


# full path to reference files
[references]
genome_fasta = /ref/Homo_sapiens.GRCh38.dna.primary_assembly.fa
genome_fastadir = /ref/fasta
genome_sizes = /ref/STAR_idx/chrNameLength.txt
genes_fasta = /ref/Homo_sapiens.GRCh38.cdna.all.fa
genes_gtf = /ref/Homo_sapiens.GRCh38.86.gtf
genes_adb = /ref/Homo_sapiens.GRCh38.86.gff3.db
genes_tsl = /ref/Homo_sapiens.GRCh38.86.gtf.tsl



# full path to program indices
[indices]
star = /ref/star_index/
bowtie = /ref/bowtie_index/hg38
starfusion = /ref/starfusion_index/
fusioncatcher = /ref/fusioncatcher_index/


# full path to program specific config files (these are just general files which need no user modification)
[other_files]
infusion_cfg = /ref/infusion.cfg
soapfuse_cfg = /ref/soapfuse.cfg
easyfuse_model = /code/easyfuse/data/model/Fusion_modeling_FFPE_train_v33.random_forest.model_full_data.EF_full.rd
```

**Comment by ibn-salem** [comment]
It seems that you are using a python interpreter from your system. Can you try: 

singularity exec 
  --containall \
  --bind </path/to/easyfuse_ref>:/ref \
  --bind </path/to/data>:/data \
  --bind </path/to/output>:/output \  
  docker://tronbioinformatics/easyfuse:latest \
  python /code/easyfuse/processing.py -i /data/ -o /output

like described in the README.

or

**python3** /code/easyfuse/processing.py -i /data/ -o /output

in the last line?

**Comment by patricksorn** [comment]
Hi @maiconfaria ,

this issue has been fixed in the latest release (1.3.7). I would close the issue.
If you encounter any problem related to this issue I would advise you to just reopen it.

Thanks for the feedback!
